Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -283,6 +330,10 @@ private RandomCutForest(Builder<?> builder, boolean notUsed) {
         centerOfMassEnabled = builder.centerOfMassEnabled;
         parallelExecutionEnabled = builder.parallelExecutionEnabled;
         compactEnabled = builder.compactEnabled;
+        singlePrecisionEnabled = builder.singlePrecisionEnabled;
+        checkArgument(!singlePrecisionEnabled || compactEnabled, "float representation requires compact trees");
```

Comment:
I think we should move this checks to the constructor (where we perform other validity checks on the builder, like checking that number of trees is positive).

Reply:
I agree with that in principle -- but right now the constructors are being called based on these flags! We need to rethink the flags as a whole.

Reply:
https://github.com/aws/random-cut-forest-by-aws/issues/82